pkg/console/subresource/configmap/configmap.go

```diff
@@ -57,22 +61,49 @@ func getApiUrl(infrastructureConfig *configv1.Infrastructure) string {
 	return ""
 }
 
-func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) *corev1.ConfigMap {
-
+// DefaultConfigMap returns
+// - a new configmap,
+// - a bool indicating if config was merged (unsupportedConfigOverrides)
+// - an error
+func DefaultConfigMap(operatorConfig *operatorv1.Console, consoleConfig *configv1.Console, infrastructureConfig *configv1.Infrastructure, rt *routev1.Route) (*corev1.ConfigMap, bool, error) {
 	logoutRedirect := getLogoutRedirect(consoleConfig)
 	brand := getBrand(operatorConfig)
 	docURL := getDocURL(operatorConfig)
 	apiServerURL := getApiUrl(infrastructureConfig)
 
 	host := rt.Spec.Host
-	config := string(NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL))
+	config := NewYamlConfig(host, logoutRedirect, brand, docURL, apiServerURL)
+
 	configMap := Stub()
-	configMap.Data = map[string]string{
-		consoleConfigYamlFile: config,
+	configMap.Data = map[string]string{}
+	unsupportedRaw := operatorConfig.Spec.UnsupportedConfigOverrides.Raw
+
+	// merge config overrides, if we have them
+	mergedConfig, err := resourcemerge.MergeProcessConfig(nil, config, unsupportedRaw)
+	if err != nil {
+		logrus.Errorf("failed to generate configmap: %v \n", err)
```

Comment:
If we were in a loop doing this many times, I would agree, but there is really not a ton of gain here since we do this once per sync loop at most.  I'm leaning towards not wanting to add additional `if` statements in `golang` `if` statment happy land....

Reply:
`failed to generate configmap` should be correct because we are `return nil, false, err`.

If the `UnsupportedConfigOverrides` are provided, but incorrect, we die, and that will bubble up and a message to fix the overrides.

Reply:
(meaning we won't generate because the function exits early)